types/react-router-redux/index.d.ts

```diff
@@ -1,63 +1,65 @@
-// Type definitions for react-router-redux 4.0
-// Project: https://github.com/rackt/react-router-redux
-// Definitions by: Isman Usoh <http://github.com/isman-usoh>
-//                 Noah Shipley <https://github.com/noah79>
-//                 Dimitri Rosenberg <https://github.com/rosendi>
-//                 Karol Janyst <https://github.com/LKay>
-//                 Dovydas Navickas <https://github.com/DovydasNavickas>
+// Type definitions for react-router-redux 5.0
+// Project: https://github.com/ReactTraining/react-router/tree/master/packages/react-router-redux
+// Definitions by: Huy Nguyen <https://github.com/huy-nguyen>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
-// TypeScript Version: 2.1
+// TypeScript Version: 2.2
 
-import { Action, Middleware, Store } from "redux";
-import { History, Location, LocationDescriptor } from "history";
+import {
+    Store,
+    Dispatch,
+    Action,
+    Middleware
+} from 'redux';
+import {
+    History,
+    Location,
+    Path,
+    LocationState,
+    LocationDescriptorObject
+} from 'history';
+import * as React from 'react';
 
-export const CALL_HISTORY_METHOD: string;
-export const LOCATION_CHANGE: string;
-
-export interface LocationActionPayload {
-    method: string;
-    args?: any[];
+export interface ConnectedRouterProps<State> {
+    store?: Store<State>,
+    history?: History
 }
+export class ConnectedRouter<State> extends React.Component<ConnectedRouterProps<State>, {}> {}
 
-export interface RouterAction extends Action {
-    payload?: LocationActionPayload;
-}
+export const LOCATION_CHANGE = '@@router/LOCATION_CHANGE'
 
-type LocationAction = (nextLocation: LocationDescriptor) => RouterAction;
-type GoAction = (n: number) => RouterAction;
-type NavigateAction = () => RouterAction;
-
-export const push: LocationAction;
-export const replace: LocationAction;
-export const go: GoAction;
-export const goBack: NavigateAction;
-export const goForward: NavigateAction;
-
-interface RouteActions {
-    push: typeof push;
-    replace: typeof replace;
-    go: typeof go;
-    goForward: typeof goForward;
-    goBack: typeof goBack;
+export interface RouterState {
+    location: Location | null
 }
 
-export const routerActions: RouteActions;
+export function routerReducer(state?: RouterState, action?: RouterAction): RouterState
 
-export interface RouterState {
-    locationBeforeTransitions: Location;
-}
+export const CALL_HISTORY_METHOD = '@@router/LOCATION_CHANGE';
+
+export function push(path: Path, state?: LocationState): RouterAction
+export function push(location: LocationDescriptorObject): RouterAction
```

Comment:
Since it's a major version update I think it's a good time to change this signature and leave only one which handles both cases. You can define it as:
```typescript
export function push(location: LocationDescriptor, state?: LocationState): RouterAction
```